### PR TITLE
fix: navigation bar color

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [MOPLAT-627] 

### Description

- moves `useAndroidNavigationBarThemeListener` to a custom hook and to the `Navigation.tsx` to get updated correctly
- fixes weird behaviors with navigationBar on android behaving weirdly

#### Demo

https://user-images.githubusercontent.com/21178754/234542492-ff14c70d-5b44-4c0c-b6d7-ca843a98f345.mov

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[MOPLAT-627]: https://artsyproduct.atlassian.net/browse/MOPLAT-627?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Navigation.tsx [hunk 1]
```diff
@@ -38,7 +39,7 @@ export const Main = () => {
   const isLoggedIn = GlobalStore.useAppState((store) => store.auth.userAccessToken) !== null
   const selectedPartner = GlobalStore.useAppState((state) => state.auth.activePartnerID)
   const isDarkMode = GlobalStore.useAppState((s) => s.devicePrefs.colorScheme === "dark")
-
+  useAndroidNavigationBarThemeListener()
```
Comment (gkartalis) [inline src/Navigation.tsx:42]: this needs to be here to change the navigationBar color correctly, before it was changing correctly only when visiting DarkModeSettings

File: src/screens/Settings/DarkModeSettings.tsx [hunk 2]
```diff
@@ -1,30 +1,16 @@
...
-        if (forcedColorScheme === "dark") {
-          await changeNavigationBarColor("black", false, false)
-        } else {
-          await changeNavigationBarColor("white", true, false)
-        }
-      }
-    }
-    update()
-  }, [isUsingSystemColorScheme, forcedColorScheme])
-
```
Comment (gkartalis) [inline src/screens/Settings/DarkModeSettings.tsx:27]: moved it to its own hook file

**Review by brainbicycle — APPROVED** [review]
🦇 🌔

File: src/screens/Settings/DarkModeSettings.tsx [hunk 3]
```diff
@@ -1,30 +1,16 @@
...
 import { Screen } from "components/Screen"
 import { SettingsItem } from "components/SettingsItem"
-import { useEffect } from "react"
-import { Platform } from "react-native"
-import changeNavigationBarColor from "react-native-navigation-bar-color"
+import { useState } from "react"
 import { GlobalStore } from "system/store/GlobalStore"
 
 export const DarkModeSettings = () => {
+  const [overrideDarkMode, setOverrideDarkMode] = useState(false)
```
Comment (damassi) [inline src/screens/Settings/DarkModeSettings.tsx:8]: @gkartalis - it's nbd, but this line might be worth a comment; like why would we want to override dark mode?

**Comment by damassi** [comment]
So tricky! tyy

**Pull request merged**